src/main/java/mate/academy/internetshop/controllers/bucket/AddProductToBucketController.java

```diff
@@ -1,31 +1,32 @@
-package mate.academy.internetshop.controllers;
+package mate.academy.internetshop.controllers.bucket;
 
 import java.io.IOException;
+import java.util.List;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import mate.academy.internetshop.lib.Injector;
 import mate.academy.internetshop.model.Bucket;
+import mate.academy.internetshop.model.Product;
 import mate.academy.internetshop.service.BucketService;
 import mate.academy.internetshop.service.ProductService;
 
 public class AddProductToBucketController extends HttpServlet {
-
     private static final Injector INJECTOR = Injector.getInstance("mate.academy.internetshop");
     private static final Long USER_ID = 1L;
-
     private BucketService bucketService = (BucketService) INJECTOR.getInstance(BucketService.class);
-
     private ProductService productService
             = (ProductService) INJECTOR.getInstance(ProductService.class);
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
+        List<Product> products = productService.getAll();
+        req.setAttribute("products", products);
+        req.getRequestDispatcher("/WEB-INF/views/products/buy.jsp").forward(req, resp);
```

Comment:
le'ts add controller(similar to GetAllProductsController) that will send user to buy.jsp and here after adding product to shoppingCart  make sendRedirect to this controller. This way you will have access to this page through it's own controller(which you can use somewhere else)